app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java

```diff
@@ -831,24 +839,39 @@ public static ReadingData parseData(int attempt, String tagId, byte[] data, Long
             int i = indexTrend - index - 1;
             if (i < 0) i += 16;
             GlucoseData glucoseData = new GlucoseData();
-            if(i * FRAM_RECORD_SIZE + TREND_START + FRAM_RECORD_SIZE >= data.length) {
+            if (i * FRAM_RECORD_SIZE + TREND_START + FRAM_RECORD_SIZE >= data.length) {
                 Log.e(TAG, "Failing to parse data from " + JoH.dateTimeText(CaptureDateTime));
                 return null;
             }
             glucoseData.glucoseLevelRaw =
                     getGlucoseRaw(new byte[]{data[(i * FRAM_RECORD_SIZE + TREND_START + 1)], data[(i * FRAM_RECORD_SIZE + TREND_START)]});
-            glucoseData.flags = LibreOOPAlgorithm.readBits(data, i * FRAM_RECORD_SIZE + TREND_START, 0xe , 0xc);
-            glucoseData.temp = LibreOOPAlgorithm.readBits(data, i * FRAM_RECORD_SIZE + TREND_START, 0x1a , 0xc);
+            glucoseData.flags = LibreOOPAlgorithm.readBits(data, i * FRAM_RECORD_SIZE + TREND_START, 0xe, 0xc);
+            glucoseData.temp = LibreOOPAlgorithm.readBits(data, i * FRAM_RECORD_SIZE + TREND_START, 0x1a, 0xc);
             glucoseData.source = GlucoseData.DataSource.FRAM;
             int time = Math.max(0, sensorTime - index);
 
             glucoseData.realDate = sensorStartTime + time * MINUTE;
             glucoseData.sensorTime = time;
-            if(verifyTime(time, "parseData trendList", data)) {
+            if (verifyTime(time, "parseData trendList", data)) {
                 trendList.add(glucoseData);
             }
         }
 
+        Collections.sort(trendList);
```

Comment:
How often is this called? Maybe a treeset is a suitable data structure, if sorting occurs frequently.

Reply:
This is only happening per reading, so probably only once in 5 minutes (or once in a minute, depend on sensor type). In any case this lists are short (around 15 items in list) and complexity of the sort is o(n log n). Inseting n objects to a tree is also n log n complexity, so I don't think that we can gain anything from it.